addons/voice/org.openhab.voice.gtts/src/main/java/org/openhab/voice/gtts/internal/GoogleTTSService.java

```diff
@@ -0,0 +1,363 @@
+        return allVoices;
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return audioFormats;
+    }
+
+    /**
+     * Helper to create AudioFormat objects from Google names.
+     *
+     * @param format Google audio format.
+     * @return Audio format object.
+     */
+    private AudioFormat getAudioFormat(String format) {
+        Integer bitDepth = 16;
+        Long frequency = 44100L;
+
+        AudioEncoding encoding = AudioEncoding.valueOf(format);
+
+        switch (encoding) {
+            case MP3:
+                // we use by default: MP3, 44khz_16bit_mono with bitrate 64 kbps
+                return new AudioFormat(AudioFormat.CONTAINER_NONE, AudioFormat.CODEC_MP3, null, bitDepth,
+                        64000, frequency);
+            case LINEAR16:
+                // we use by default: wav, 44khz_16bit_mono
+                return new AudioFormat(AudioFormat.CONTAINER_WAVE, AudioFormat.CODEC_PCM_SIGNED, null, bitDepth,
+                        null, frequency);
+            default:
+                logger.warn("Audio format {} is not yet supported.", format);
+                return null;
+        }
+    }
+
+    /**
+     * Checks parameters and calls the API to synthesize voice.
+     *
+     * @param text            Input text.
+     * @param voice           Selected voice.
+     * @param requestedFormat Format that is supported by the target sink as well.
+     * @return Output audio stream
+     * @throws TTSException in case the service is unavailable or a parameter is invalid.
+     */
+    @Override
+    public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFormat) throws TTSException {
+        logger.debug("Synthesize '{}' for voice '{}' in format {}", text, voice.getUID(), requestedFormat);
+        // Validate known api key
+        if (!apiImpl.isInitialized()) {
+            throw new TTSException("Missing service configuration.");
+        }
+        // Validate arguments
+        // trim text
+        text = text.trim();
+        if (text.isEmpty()) {
+            throw new TTSException("The passed text is null or empty");
+        }
+        if (!this.allVoices.contains(voice)) {
+            throw new TTSException("The passed voice is unsupported");
+        }
+        boolean isAudioFormatSupported = false;
+        for (AudioFormat currentAudioFormat : this.audioFormats) {
+            if (currentAudioFormat.isCompatible(requestedFormat)) {
+                isAudioFormatSupported = true;
+                break;
+            }
+        }
+        if (!isAudioFormatSupported) {
+            throw new TTSException("The passed AudioFormat is unsupported");
+        }
+
+        // now create the input stream for given text, locale, format. There is
+        // only a default voice
+        try {
+            File audioFile = apiImpl.synthesizeSpeech(text, new GoogleTTSVoice(voice), requestedFormat.getCodec());
```

Comment:
What's the reason that you are creating a copy of the voice instance here? Seems to be unnecessary as a voice instance is supposed to be immutable.

Reply:
Done